### PR TITLE
Symbol Signature text API

### PR DESCRIPTION
**Pull request by nojaf**
Related to https://github.com/dotnet/fsharp/issues/13215 and https://github.com/dotnet/fsharp/issues/14170

I want to add a new API to `FSharpMemberOrFunctionOrValue` to return the "signature text" via the `NicePrint` API. This would be useful for editors to provide a light bulb kind of action to add a member or binding to an existing signature file.

I'm not quite sure yet about the name or exact API but I'm very happy with the initial results of this. It is quite accurate and has a lot of details right. 

For example:
```fsharp
let f (a : string) (b : char) = fun (c : float) -> 2.0
```
gives for `f`
```fsharp
val f: a: string -> b: char -> c: float -> float
```
Or
```fsharp
type State =
    {
        Files : string list
    }

    static member Empty : State = { Files = [] }
```
gives for `Empty`
```fsharp
static member Empty: State
```

Let me know what you think.

### CODE REVIEW

**Comment by T-Gro** [comment]
Do you have a real use case in mind already?
For the editor tooling now (closest coming to my mind for this feature are mouseover tooltips), TaggedText is preferred over a plain string - makes more sense given requirements for coloring or for example hyperlinks.

**Comment by nojaf** [comment]
> Do you have a real use case in mind already?

Adding a member or binding to an existing signature file. Or updating an outdated one.

**Review by T-Gro — APPROVED** [review]

**Comment by psfinaki** [comment]
Might be useful for signature hints. Will look at this in the upcoming days - thanks!

**Comment by nojaf** [comment]
This API is working out very well for me in Telplin. (Proof of concept in https://github.com/nojaf/telplin/pull/77).

What Telplin does, is traverse the untyped tree to generate a signature counterpart.
Each time it needs typed information, it fetches this for the relevant `Symbol`.
Invokes the `GetValSignatureText` call to get the signature-related text.
It parses this and converts it to the Fantomas tree model and inserts it in the signature tree.

The `DisplayEnvironment` for the `Symbol` is much better than what `FSharpCheckFileResults,GenerateSignature` starts with, and since `open statements` are just copied over from the untyped tree, this works out beautifully. Any details that are missed by `GetValSignatureText` call (like https://github.com/dotnet/fsharp/issues/15267 for example) can be updated from the untyped tree information. These short-term workarounds should of course be fixed eventually in NicePrint, but it allows for a convenient escape hatch.

As mentioned in the OP, I have incremental editor scenarios in mind as well for this.
VSCode would nearly instantly benefit from this API to add their light bubble action thing.

To summarize, can we have this, please?

**Review by 0101 — APPROVED** [review]

File: src/Compiler/Symbols/Symbols.fsi [hunk 1]
```diff
@@ -966,6 +966,9 @@ type FSharpMemberOrFunctionOrValue =
     /// Format the type using the rules of the given display context
     member GetReturnTypeLayout: displayContext: FSharpDisplayContext -> TaggedText[] option
 
+    /// Get the signature text to included this Symbol into an existing signature file.
```
Comment (psfinaki) [inline src/Compiler/Symbols/Symbols.fsi:969]:
```suggestion
    /// Get the signature text to include this Symbol into an existing signature file.
```

**Review by psfinaki — APPROVED** [review]
Looks good, thanks!

**Comment by nojaf** [comment]
Thanks all!
